### PR TITLE
Refactor attributeValues resolvers in translations to use dataloaders

### PR DESCRIPTION
**Pull request by dexon44**
I want to merge this change because it refactor attributeValues resolvers in translations to use dataloaders.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/translations/types.py [hunk 1]
```diff
@@ -146,13 +151,20 @@ def resolve_product_variant(root: product_models.ProductVariant, info):
...
     @staticmethod
     def resolve_attribute_values(root: product_models.ProductVariant, _info):
-        translatable_values = []
-        for assignment in root.attributes.all():
-            attr = assignment.attribute
-            if attr.input_type in AttributeInputType.TRANSLATABLE_ATTRIBUTES:
-                value = assignment.values.first()
-                translatable_values.append(value)
-        return translatable_values
+        def func(attributes):
```
Comment (maarcingebala) [inline saleor/graphql/translations/types.py:154]: `func` is a good name when you're building a draft of some changes; here it should be renamed to something more understandable by others. `get_translatable_attributes` for example?

Comment (maarcingebala) [reply saleor/graphql/translations/types.py:154]: This function is identical in all three resolvers, we could actually extract it outside the class and reuse it. Adding a short docstring would be then useful as well.

Comment (tomaszszymanski129) [reply saleor/graphql/translations/types.py:154]: Usually when dealing with single dataloader in resolvers, we compose func name like underscore + resolver name. For ex. `_resolve_attribute_values`. But in that case, yes, it should be extracted (and no underscore then).

File: saleor/graphql/translations/types.py [hunk 2]
```diff
@@ -146,13 +151,20 @@ def resolve_product_variant(root: product_models.ProductVariant, info):
...
+            translatable_values = []
+            for assignment in attributes:
+                attr = assignment["attribute"]
+                if attr.input_type in AttributeInputType.TRANSLATABLE_ATTRIBUTES:
+                    value = assignment["values"][0]
+                    translatable_values.append(value)
+            return translatable_values
+
+        return (
+            SelectedAttributesByProductVariantIdLoader(_info.context)
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/types.py:164]:
```suggestion
            SelectedAttributesByProductVariantIdLoader(info.context)
```
info is no longer unused so remove `_` from it

File: saleor/graphql/translations/types.py [hunk 3]
```diff
@@ -29,6 +34,20 @@
 ]
 
 
+def get_translatable_attribute_values(attributes: list) -> list:
```
Comment (tomaszszymanski129) [inline saleor/graphql/translations/types.py:37]: Can we do `List[type]` instead of generic `list`?

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**